src/java/org/apache/cassandra/index/sai/disk/v1/SkinnyRowAwarePrimaryKeyMap.java

```diff
@@ -109,89 +101,71 @@ public RowAwarePrimaryKeyMapFactory(IndexDescriptor indexDescriptor, SSTableRead
+            return new SkinnyRowAwarePrimaryKeyMap(rowIdToToken,
+                                                   sortedTermsReader.openCursor(),
+                                                   partitioner,
+                                                   primaryKeyFactory);
         }
 
         @Override
         public void close()
         {
-            FileUtils.closeQuietly(Arrays.asList(tokensFile, primaryKeyBlocksFile, primaryKeyBlockOffsetsFile, primaryKeyTrieFile));
+            FileUtils.closeQuietly(Arrays.asList(tokensFile, primaryKeyBlocksFile, primaryKeyBlockOffsetsFile));
         }
     }
 
-    private final LongArray rowIdToToken;
-    private final TriePrefixSearcher triePrefixSearcher;
-    private final SortedTermsReader.Cursor cursor;
-    private final IPartitioner partitioner;
-    private final PrimaryKey.Factory primaryKeyFactory;
-    private final ClusteringComparator clusteringComparator;
-    private final ByteBuffer tokenBuffer = ByteBuffer.allocate(Long.BYTES);
-
-    private RowAwarePrimaryKeyMap(LongArray rowIdToToken,
-                                  TriePrefixSearcher triePrefixSearcher,
-                                  SortedTermsReader.Cursor cursor,
-                                  IPartitioner partitioner,
-                                  PrimaryKey.Factory primaryKeyFactory,
-                                  ClusteringComparator clusteringComparator)
+    protected final LongArray tokenArray;
+    protected final SortedTermsReader.Cursor cursor;
+    protected final IPartitioner partitioner;
+    protected final PrimaryKey.Factory primaryKeyFactory;
+    protected final ByteBuffer tokenBuffer = ByteBuffer.allocate(Long.BYTES);
+
+    protected SkinnyRowAwarePrimaryKeyMap(LongArray tokenArray,
+                                          SortedTermsReader.Cursor cursor,
+                                          IPartitioner partitioner,
+                                          PrimaryKey.Factory primaryKeyFactory)
     {
-        this.rowIdToToken = rowIdToToken;
-        this.triePrefixSearcher = triePrefixSearcher;
+        this.tokenArray = tokenArray;
         this.cursor = cursor;
         this.partitioner = partitioner;
         this.primaryKeyFactory = primaryKeyFactory;
-        this.clusteringComparator = clusteringComparator;
     }
 
     @Override
     public PrimaryKey primaryKeyFromRowId(long sstableRowId)
     {
-        tokenBuffer.putLong(rowIdToToken.get(sstableRowId));
+        tokenBuffer.putLong(tokenArray.get(sstableRowId));
         tokenBuffer.rewind();
         return primaryKeyFactory.createDeferred(partitioner.getTokenFactory().fromByteArray(tokenBuffer), () -> supplier(sstableRowId));
     }
 
     @Override
     public long rowIdFromPrimaryKey(PrimaryKey key)
     {
-        return triePrefixSearcher.prefixSearch(key.asComparableBytes(ByteComparable.Version.OSS50));
+        return tokenArray.findTokenRowID(key.token().getLongValue());
```

Comment:
Just to refresh my memory, this never had to be exact, because it's only used for skipping, right?

Reply:
I'm not sure what you mean by `exact` here, but, yes, it is only used for skipping so only needs to get to the first matching token.